hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -868,7 +889,7 @@ CompletableFuture<Void> onParticipantGracefulShutdown(UUID uuid) {
     @Nonnull
     CompletableFuture<Void> gracefullyTerminate() {
         CompletableFuture<CompletableFuture<Void>> future = mc.coordinationService().submitToCoordinatorThread(
-                () -> requestTermination(RESTART_GRACEFUL, false).f0());
+                () -> requestTermination(RESTART_GRACEFUL, false, true).f0());
```

Comment:
I intended`userInitiatedTermination` flag to be more general (not only for cancel) but currently it is needed only for cancel.
I was thinking about clearing `userInitiatedTermination` when we clear `requestedTerminationMode` in `finalizeJob` and the other place (I thought it would be redundant, but as you shown - it will not). These fields together make full information about single (current) job termination request.

Reply:
`requestedTerminationMode` and `userInitiatedTermination` are now part of single, immutable `TerminationRequest` which is cleared after execution is terminated